portfolio/src/main/webapp/script.js

```diff
@@ -14,6 +14,13 @@
 
 async function getData() {
     const response = await fetch('/data');
-    const data = await response.text();
-    document.getElementById('data-container').innerHTML = data;
+    const comments = await response.json();
+    let html = "";
+    for(const comment of comments) {
+        html += "<div id=\"comment\">\n";
```

Comment:
I think you might want `class` instead of `id` here. If you have multiple elements with the same `id`, your browser will start to yell at you.